controllers/cloud.redhat.com/metrics.go

```diff
@@ -0,0 +1,51 @@
+package controllers
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	totalSuccessfulPoolReservationsCountMetrics = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "total_successful_pool_reservation_counter",
+			Help: "Total successful reservations from each pool",
+		},
+		[]string{"pool"},
+	)
+
+	totalFailedPoolReservationsCountMetrics = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "total_failed_pool_reservation_counter",
+			Help: "Total failed reservations from each pool",
+		},
+		[]string{"pool"},
+	)
+
+	averageRequestedDurationMetrics = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "average_duration_for_namespace_reservations_in_hours",
```

Comment:
This one is almost perfect, just a bit verbose imo.

Reply:
I went with `average_time_reservation_to_deployment_milliseconds`- let me know if you're thinking something better :)